### PR TITLE
management document.

### PR DESCRIPTION
**Pull request by ydye**
Not Finished Yet.

### CODE REVIEW

File: pai-management/doc/how-to-setup-dev-box.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+```
+sudo apt-get install python python-paramiko python-yaml python-jinja2
+sudo pip install python-etcd kubernetes
+
+# clone our repo first.
+git clone https://github.com/Microsoft/pai.git
+
+# Go into the workdir.
+cd pai/pai-management/
+```
```
Comment (fanyangCS) [inline pai-management/doc/how-to-setup-dev-box.md]: I suggest we remove this option.

Comment (fanyangCS) [inline pai-management/doc/how-to-setup-dev-box.md]: I suggest we remove the support to host env for this release. just focus on docker env.

File: pai-management/README.md [hunk 2]
```diff
@@ -17,332 +17,40 @@
...
 
-If your environment don't have a load-balance service. You could add proxy node to the kuebrnetes cluster. So you will have to add a node with the k8s-role as proxy. Now we only support single node proxy. If you want to deploy a proxy with ha, you could implement it by yourself.
+[A Guide For Build Service Image](doc/image-build.md)
 
-[The proxy component definition](k8sPaiLibrary/maintainconf/deploy.yaml)
+[A Guide For Service Maintenance](doc/service-maintain.md)
 
-[The component templatefile path](k8sPaiLibrary/template)
+TODO
+[A Guide For Pulling Image For the Public Registry]
```
Comment (fanyangCS) [inline pai-management/README.md]: >For [](start = 27, length = 3)

from

File: pai-management/README.md [hunk 3]
```diff
@@ -17,332 +17,40 @@
...
+#### Cluster Bootstrap
 
-#### solution 1
+TODO
 
-Because cloud providers such as azure always have the load balance service. So when deploy pai to the cloud platform, you could chose the load-balance service to implement the high-availability.
+#### Machine (Kubernetes) Maintenance
 
-Before bootstrap your kubernetes cluster, you should configure your load-balance. please set the backend with your master. And set the following property with the ip of the load-balance in the [kubernetes-configuration.yaml](https://github.com/Microsoft/pai/blob/master/cluster-configuration/kubernetes-configuration.yaml).
+- Please maintain your machine in the dev-box
```
Comment (fanyangCS) [inline pai-management/README.md]: >Please maintain your machine in the dev-box [](start = 2, length = 43)

do you mean: please maintain a pai cluster from within a dev-box?

File: pai-management/README.md [hunk 4]
```diff
@@ -17,332 +17,40 @@
...
-Note: Don't change the file name!!!!!!!!!!!!
 
-## Kubernetes high-availability
+#### Cluster Bootstrap
 
-#### solution 1
+TODO
 
-Because cloud providers such as azure always have the load balance service. So when deploy pai to the cloud platform, you could chose the load-balance service to implement the high-availability.
+#### Machine (Kubernetes) Maintenance
```
Comment (fanyangCS) [inline pai-management/README.md]: >Machine (Kubernetes) [](start = 5, length = 20)

PAI Machine Maintenance

File: pai-management/README.md (hunk 3 above)
Comment (fanyangCS) [inline pai-management/README.md]: >Please maintain your machine in the dev-box [](start = 2, length = 43)

Please maintain the PAI cluster from within a dev-box described above.

File: pai-management/README.md [hunk 5]
```diff
@@ -17,332 +17,40 @@
...
 
-Before bootstrap your kubernetes cluster, you should configure your load-balance. please set the backend with your master. And set the following property with the ip of the load-balance in the [kubernetes-configuration.yaml](https://github.com/Microsoft/pai/blob/master/cluster-configuration/kubernetes-configuration.yaml).
+- Please maintain your machine in the dev-box
 
-```yaml
+[A Guide For Machine(Kubernetes) Maintenance](doc/machine-maintenance.md)
```
Comment (fanyangCS) [inline pai-management/README.md]: >Machine(Kubernetes) [](start = 13, length = 19)

PAI Machine

File: pai-management/doc/how-to-setup-dev-box.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+cd pai/pai-management/
+
+# Build your dev-box.
+sudo docker build -t dev-box .
+
+```
+
+#### Starting Your Dev-Box's contianer
+
+- Suppose the path of ```custom-hadoop-binary-path``` in your service-configuration is ```/pathHadoop```
```
[the remark this answers is not in the document]
Comment (ydye) [reply pai-management/doc/how-to-setup-dev-box.md]: We'd better ask user to do this.

File: pai-management/doc/how-to-setup-dev-box.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+- Suppose the directory path of your cluster-configuration is ``/pathConfiguration````. Note: Don't change the configuration file name！
```
[the remark this answers is not in the document]
Comment (ydye) [reply pai-management/doc/how-to-setup-dev-box.md]: We'd better not.

**Comment by ydye** [comment]
@hwuu @wangcan0329 @sterowang @fanyangCS 
Example to add service is finished. Follow the guide, user will learn from the example of hbase to add their service. Some tutorial about cluster-object-model is left. And this part will finish when releasing C-O-M.

**Pull request merged**